cli/pkg/local/app.go

```diff
@@ -182,9 +184,12 @@ func NewApp(ctx context.Context, opts *AppOptions) (*App, error) {
 		"rill.download_row_limit": "0", // 0 means unlimited
 		"rill.stage_changes":      "false",
 	}
-	for k, v := range opts.Variables {
-		vars[k] = v
+	maps.Copy(vars, opts.Variables)
+	envMap, err := godotenv.Read(filepath.Join(projectPath, ".env"))
+	if err != nil && !errors.Is(err, os.ErrNotExist) {
+		return nil, err
 	}
+	maps.Copy(vars, envMap)
```

Comment:
**Scalability Issue**: The use of godotenv.Read to load environment variables from a .env file during runtime can lead to scalability issues. This approach does not support horizontal scaling well because each instance would need to manage its own set of environment variables independently, potentially leading to inconsistencies across instances. <br> **Fix**: Implement a centralized configuration management service to handle environment variables. This service can distribute configuration updates in real-time to all instances, ensuring consistency across the application's environment. <br> **Code Suggestion**: 
 ```
 -	envMap, err := godotenv.Read(filepath.Join(projectPath, ".env"))
 -	if err != nil && !errors.Is(err, os.ErrNotExist) {
 -		return nil, err
 -	}
 -	maps.Copy(vars, envMap)
 +	configService, err := config.NewCentralConfigService()
 +	if err != nil {
 +		return nil, fmt.Errorf("failed to initialize configuration service: %w", err)
 +	}
 +	err = configService.LoadEnvironmentVariables(projectPath, vars)
 +	if err != nil {
 +		return nil, fmt.Errorf("failed to load environment variables: %w", err)
 +	}
 ```